genesis/example/allowance.go

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright 2018 INS Ecosystem
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package example
+
+import (
+	"fmt"
+
+	"github.com/insolar/insolar/genesis/model/class"
+	"github.com/insolar/insolar/genesis/model/contract"
+	"github.com/insolar/insolar/genesis/model/object"
+)
+
+type Allowance interface {
+	object.Composite
+	contract.SmartContract
+	GetAmount() int
+	GetSender() string
```

Comment:
You need to store both sender and recipient

Reply:
You need an operation that marks Allowance as completed.

Reply:
Why sender is string?

Reply:
About recipient: I add it, but it's not pretty clear now if we need it. Since, allowance may locate inside recipient, and that is why we might not need this field